src/main/java/com/google/cloud/resourcemanager/OperationInfo.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.cloudresourcemanager.model.Operation;
+import com.google.common.base.MoreObjects;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/** Google Resource Manager Operation metadata. */
+public class OperationInfo {
+
+  private final Boolean done;
+  private final Status error;
+  private final Map<String, Object> metadata;
+  private final String name;
+  private final Map<String, Object> response;
```

Comment:
ditto

Reply:
These fields come as a batch and we set those in protobuf to send for RPC call. Also, the protobuf fields expect Map<String, Object> so after creating a custom class, we will have to use Map as we need to convert collected data for protobuf. see https://github.com/googleapis/google-api-java-client-services/blob/master/clients/google-api-services-cloudresourcemanager/v1/1.30.1/com/google/api/services/cloudresourcemanager/model/Operation.java#L77